### ISSUE TITLE
TextEntry Weird Behavior (pyglet 2.0.3)

### ISSUE THREAD

**Issue by onuelito** [pyglet/pyglet#749]
**Description**
Writing a text in the `pyglet.gui.TextEntry(*args, **kwargs)` object that exceeds it's width parameter makes it have the weird behavior of offsetting the text within it.

**System Requirements**
I don't it is related to the System/OS as I have tried on my Linux Machine and as well as my Windows one that is completely different and it does the same thing. I think it is more related to the source code. Regardless, here are my PC info : 

```
Graphics Card : NVIDIA GeForce RTX 3060
Processor : Amd Ryzen 1600 Six-Core
Operating System : Arch-Linux
Audrio Driver : OpenAL
```

**How to reproduce**
Here is the code with which I discovered it : 
```py
import pyglet
import pyglet.gl as gl

window = pyglet.window.Window(
    640, 480, caption="Contribution"
)

batch = pyglet.graphics.Batch()
entry = pyglet.gui.TextEntry("",100, 100, 200,batch=batch)
window.push_handlers(entry)

@window.event
def on_draw():
    gl.glClearColor(0.7, 0.7, 0.7, 1.0)
    window.clear()
    batch.draw()


pyglet.app.run()
```

https://user-images.githubusercontent.com/90167681/211968311-371abded-f36b-44ac-8df9-1faf91d68c5e.mp4

(This issue is not present in pyglet 1.5.27)

**Comment by shenjackyuanjie** [comment]
still exist in 2.0.5

**Comment by caffeinepills** [comment]
https://github.com/caffeinepills/pyglet/tree/ti_fix 

Try my branch and let me know if it is any better.

**Comment by onuelito** [comment]
I tested it a couple of times. For the most part it works it's just that sometimes it will break as seen in this clip around `00:20` seconds : 

https://user-images.githubusercontent.com/90167681/222308087-1d1050c7-9330-40c5-9173-653dbd89e21f.mp4

**Comment by caffeinepills** [comment]
I can't replicate that issue on my end.

**Comment by onuelito** [comment]
I have no clue of how to find a pattern to repeat the issue consistently as I just tried random stuff while testing it multiple times until it just happened (I just typed and erased at different points). Maybe this will help : I installed the current version of pyglet (right window) in a virtual environment while testing your patch (left window) and noticed that the caret highlight was different in both : 

https://user-images.githubusercontent.com/90167681/222596900-29402b64-2a17-452c-87eb-b729a5396cc2.mp4

**Comment by caffeinepills** [comment]
Can you try updating my branch and see if it's any better?

**Comment by caffeinepills** [comment]
Closed as resolved in #786
